Userland/Libraries/LibBitTorrent/Bencode/BTypes.h

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2023, Pierre Delagrave <pierre.delagrave@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/DeprecatedString.h>
+#include <AK/HashMap.h>
+#include <AK/Variant.h>
+
+namespace BitTorrent {
+
+struct List;
+class Dict;
```

Comment:
This is a bit more complicated for `Dict` but:

1. No need for a forward declaration
2. We usually prefer to have the map as a member over inheriting it
3. Is really worth it? As far as I have seen, that's a lot of lines only to prevent a few `.value().get<T>()`

Reply:
1. I would like to learn how to avoid it, so far it seems necessary with such recursive definitions.
2. A dict is a synonym for a map, I just wanted to reuse the wording of the spec. `BitTorrent::Dict` hence is a Map with a few helper methods to help keep the code more readable. Unsure about the benefits of using composition in that case.
3. There's 22 usages (`Announcer` and `MetaInfo`) and it does make the code more readable, at least on par if not more than this change: https://github.com/SerenityOS/serenity/pull/20022#discussion_r1265841856 , for comparison.